src/main/java/org/mariadb/jdbc/internal/com/read/dao/CmdInformationBatch.java

```diff
@@ -95,6 +95,16 @@ public void addErrorStat() {
         this.updateCounts.add((long) Statement.EXECUTE_FAILED);
     }
 
+    /**
+     * Clear error state, used for clear exception after first batch query, when fall back to per-query execution.
+     *
+     */
+    @Override
+    public void clearErrorStat() {
+        hasException = true;
```

Comment:
reverting must then set hasException to false

Reply:
My mistake, of course this should be set to false!